src/test/java/io/vertx/test/core/SchedulerInterceptorTest.java

```diff
@@ -0,0 +1,167 @@
+package io.vertx.test.core;
+
+import org.junit.Test;
+
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.BiFunction;
+
+import io.vertx.core.Context;
+import io.vertx.core.Vertx;
+
+/**
+ * Unit-Tests of {@link Vertx} scheduler interceptor behavior
+ */
+public class SchedulerInterceptorTest extends VertxTestBase {
+
+  @Test
+  public void testRunOnContext() {
+    AtomicInteger cnt = new AtomicInteger();
+    vertx.addSchedulerInterceptor((c, r) -> {
+      cnt.incrementAndGet();
+      return r;
+    });
+
+    vertx.runOnContext(v -> {
+      assertEquals(1, cnt.get());
+    });
+    vertx.runOnContext(v -> {
+      assertEquals(2, cnt.get());
+      testComplete();
```

Comment:
A very similar test with vertx.setTimer and vertx.setPeriodic and vertx.setPeriodicStream would be beneficial.